backend/populatedb.js

```diff
@@ -0,0 +1,21 @@
+var mongoose = require('mongoose');
+var User = require('./models/user');
+
+const mongoConnString = process.env.MONGODB_CONNSTRING ? process.env.MONGODB_CONNSTRING : 'mongodb://pharmacy-information-root:pharmacy-information-password@localhost:27017?retryWrites=true&w=majority';
+mongoose.connect(mongoConnString, {useNewUrlParser: true , useUnifiedTopology: true});
+
+const user = new User({
+  name: 'alice',
+  contact: '555-555-5555',
+  nic: '1234',
+  email: 'alice@example.com' ,
+  password: '$2a$10$CMRf90.VvXfd9CN04duIi.QvtO.US1Gcdzs3aRJsbWztKBstm8wgy',
+  role: 'pharmacist',
+});
+
+// If we don't disable validation, mongoose-unique-validator will erroneously throw a ValidationError
+// on the User email unique constraint
+user.save({ validateBeforeSave: false })
```

Comment:
>wouldn't it only complain about validation on the unique email constraint if an entry with that email already exists?

I was getting the validation error even when there were zero existing users (hence the "erroneously" in my comment). I couldn't figure out why it was happening, but if you want to try removing the `validateBeforeSave: false` in your testing, please let me know if you get the same result.

>it seems like this save function is executed every time the container starts up based on the docker file cmd and thus resulting in multiple repeat entries into the users mongo collection with validation disabled.

Good point. I'll add a check for existing user. But I still think I'll have to disable validation due to the issue I described.

Reply:
I did check that the database was empty when I got the seemingly erroneous duplicate email error, but after doing this refactor, I cannot reproduce anymore 🤷. Note, however, that I decided not to add a check for if the user exists already because I don't think we really get any benefit from doing so; if the user already exists when the container starts up, it will simply throw the correct validation error and continue starting up the application. https://github.com/mcode/pharmacy-information-system/pull/11/commits/636459aead979959170b0d5b5d18c77810bab835